scripts/check_phase42_boundary_correspondence.py

```
@@ -0,0 +1,1074 @@
+
+def require_hash32(label: str, value: Any) -> None:
+    if not isinstance(value, str) or not HASH32_RE.fullmatch(value):
+        raise Phase42Error(f"{label}: expected a 32-byte lowercase hex commitment")
+
+
+def require_bool(label: str, value: Any) -> None:
+    if not isinstance(value, bool):
+        raise Phase42Error(f"{label}: expected a boolean")
+
+
+def require_usize(label: str, value: Any) -> int:
```

Ruff (0.15.10)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)